exercises/concept/ellens-alien-game/.docs/introduction.md

```diff
@@ -0,0 +1,104 @@
+# Introduction
+
+## Classes
+
+It is time to get to one of the core paradigms of C++: object-oriented programming (OOP).
+OOP is centered around `classes` - user-defined types of data with their own set of related functions.
+We will start with the basics and will cover more advanced topics further down the syllabus tree.
+
+### Members
+
+Classes can have **member variables** and **member functions**.
+They are accessed by the **member selection** operator `.`.
+Just as variables outside of `classes`, it is advisable to initialize member variables with a value upon declaration.
+This value will then become the default for newly created objects from this class.
+
+### Encapsulation and Information Hiding
+
+Classes offer the option to restrict access to their members.
+The two basic cases are `private` and `public`.
+`private` members are not accessible from outside the class.
+`public` members can be called freely.
+All members are `private` by default and need to be explicitly marked to be usable outside of the class.
+
+### Basic example
+
+The definition of a new `class` can be seen in the following example.
+Notice the `;` after the definition.
+
+```cpp
+class Wizzard {
+  public:               // from here on all members are accessible publicly
+    inc cast_spell() {  // defines the public member function cast_spell
+      return damage;
+    }
+    std::string name{}; // defines the public member variable `name`
+  private:              // from here on all members are private
+    int damage{5};      // defines the private member variable `damage`
+};
+
+```
+
+You can access all variables from within the class.
+Take a look at `damage` inside the `cast_spell` function.
+You cannot read or change `private`  members outside of the class:
+
+```cpp
+Wizzard silverhand{};
+// calling the `cast_spell` function is okay, it is public:
+iggwilv.cast_spell();
+// => 5
+// name is public and can be changed:
+silverhand.name = "Laeral";
+// damage is private -> Compilation error
+silverhand.damage = 500; 
```

Comment:
everywhere else we have:
some_statement;
// => expected result

here the order is reversed:
// explanation -> expected error
some_statement; 

Instead:
silverhand.damage = 500;
// damage is private => Compilation error